app/main.py

```diff
@@ -1 +1,30 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        print(f"Animal __init__ called with: {name}, health: {health}")
+        self.name = name
+        self.health = health
+        self.hidden = False
+        Animal.alive.append(self)
+
+    def die(self) -> None:
+        if self in Animal.alive:
+            Animal.alive.remove(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
```

Comment:
Code Efficiency issue: Use the class name to get access to the class attributes. In this case, you should use `Herbivore.hidden` instead of `self.hidden`. Here is the corrected code: `Herbivore.hidden = not Herbivore.hidden`.

Reply:
done